firebase_admin/__init__.py

```diff
@@ -215,39 +215,15 @@ def __init__(self, name, credential, options):
         self._options = _AppOptions(options)
         self._lock = threading.RLock()
         self._services = {}
-        self._project_id = App._lookup_project_id(self._credential, self._options)
 
-    @classmethod
-    def _lookup_project_id(cls, credential, options):
-        """Looks up the Firebase project ID associated with an App.
-
-        This method first inspects the app options for a ``projectId`` entry. Then it attempts to
-        get the project ID from the credential used to initialize the app. If that also fails,
-        attempts to look up the ``GOOGLE_CLOUD_PROJECT`` and ``GCLOUD_PROJECT`` environment
-        variables.
-
-        Args:
-            credential: A Firebase credential instance.
-            options: A Firebase AppOptions instance.
-
-        Returns:
-            str: A project ID string or None.
+        App._validate_project_id(self._options.get('projectId'))
+        self._project_id_initialized = False
 
-        Raises:
-            ValueError: If a non-string project ID value is specified.
-        """
-        project_id = options.get('projectId')
-        if not project_id:
-            try:
-                project_id = credential.project_id
-            except AttributeError:
-                pass
-        if not project_id:
-            project_id = os.environ.get('GOOGLE_CLOUD_PROJECT', os.environ.get('GCLOUD_PROJECT'))
+    @classmethod
+    def _validate_project_id(cls, project_id):
         if project_id is not None and not isinstance(project_id, six.string_types):
             raise ValueError(
                 'Invalid project ID: "{0}". project ID must be a string.'.format(project_id))
-        return project_id
```

Comment:
We did not raise an exception when project_id is the empty string `""`, e.g. In cases like `options={projectId: ""}`. Is this intentional? Or shall I add a check for it? Or shall I normalize it to `None`?

Reply:
Not really intentional. I suppose because of how our lookup logic is implemented, `project_id` is empty only if the corresponding env variables are set to the empty string. We can continue to ignore that case, or if you like go ahead and normalize to None.

Reply:
Okay, I'm continuing to ignore that case.